src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -107,6 +109,36 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
             return Ok(status);
         }
 
+        [HttpGet]
+        [Route("admin/host/workerProcesses")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> GetWorkerProcesses([FromServices] IServiceProvider serviceProvider = null)
+        {
+            var processes = new List<WorkerProcessInfo>();
+            // When making changes to HostStatus, ensure that you update the HostStatus class in AAPT-Antares-Websites repo as well.
+            var rpcFunctionInvocationDispatcher = serviceProvider.GetService<RpcFunctionInvocationDispatcher>();
+            if (rpcFunctionInvocationDispatcher != null)
+            {
+                var channels = await rpcFunctionInvocationDispatcher.GetAllWorkerChannelsAsync();
+                if (channels != null)
+                {
+                    foreach (var channel in channels)
+                    {
+                        var processInfo = new WorkerProcessInfo()
+                        {
+                            ProcessId = channel.Id,
+                            ExeName = channel.WorkerProcess.Process.StartInfo.FileName,
+                            DebugEngine = "python",
+                            IsEligibleForOpenInBrowser = false
+                        };
+                        processes.Add(processInfo);
+                    }
+                }
+            }
+
+            return Ok(processes);
```

Comment:
Could something go wrong here where we might need to be able to return a different http code?

Reply:
Updated the code to return 503 if we are not able to get required services. Otherwise we would simply return empty collection